test/end2end_test.go

```diff
@@ -7668,3 +7668,82 @@ func (s) TestClientSettingsFloodCloseConn(t *testing.T) {
 	s.GracefulStop()
 	timer.Stop()
 }
+
+// TestDeadlineSetOnConnectionOnClientCredentialHandshake tests that there is a deadline
+// set on the net.Conn when a credential handshake happens in http2_client.
+func (s) TestDeadlineSetOnConnectionOnClientCredentialHandshake(t *testing.T) {
+	lis, err := net.Listen("tcp", "localhost:50053")
+	if err != nil {
+		t.Fatalf("Failed to listen: %v", err)
+	}
+	go func() {
+		lis.Accept()
+	}()
+	credsInvoked := testutils.NewChannel()
+	cvd := &credentialsVerifyDeadline{
+		credsInvoked: credsInvoked,
+	}
+	dopts := []grpc.DialOption{
+		grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
+			conn, err := (&net.Dialer{}).DialContext(ctx, "tcp", addr)
+			if err != nil {
+				return nil, err
+			}
+			cpd := &connPersistDeadline{
+				Conn: conn,
+			}
+			return cpd, nil
+		}),
+		grpc.WithTransportCredentials(cvd),
+	}
+	cc, err := grpc.Dial(lis.Addr().String(), dopts...)
+	if err != nil {
+		t.Fatalf("Failed to dial: %v", err)
+	}
+	defer cc.Close()
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	connectionHasCorrectDeadline, err := credsInvoked.Receive(ctx)
+	if err != nil {
+		t.Fatalf("Error receiving from credsInvoked: %v", err)
+	}
+	if !connectionHasCorrectDeadline.(bool) {
+		t.Fatalf("Connection did not have deadline set.")
+	}
+}
+
+type connPersistDeadline struct {
```

Comment:
`Persist` to me implies it lasts forever.  I can't come up with a good name for this, though.  `deadlineConn` is probably too vague as well.  Maybe `getDeadlineConn` or `infoConn` (which could be extended to store other info about the conn as needed) or `storeDeadlineConn`?

(Also as a nit: note that the usual naming pattern we follow is either `somethingSomethingConn` or `connWithSomething`.)

Reply:
I like infoConn. I want this test struct to provide utility past this specific fixit issue.